File: Formula1/src/Formula1/JAVA/F3.java
package Formula1.JAVA;

public class F3 extends Monoplaza{
    private String nombreAcademia;

    ////////////////CONSTRUCTOR/////////////
    public F3(String nombrePiloto, String nacionalidad, int numeroPlaza, String escuderia, int cantidadPuntos, String pNombreAcademia) {
        super(nombrePiloto, nacionalidad, numeroPlaza, escuderia, cantidadPuntos);
        this.nombreAcademia = pNombreAcademia;
    }
    ///////////////////GETTERS Y SETTERS/////////////////
    public void setNombreAcademia(String nombreAcademia) {
        this.nombreAcademia = nombreAcademia;
    }
    public String getNombreAcademia() {
        return nombreAcademia;
    }

    ////////////METODOS////////////////
    @Override
public boolean posicionValida(int posicion) {
    if(posicion >=1 && posicion<=30){
        return true;
    }else{
        return false;
    }
}


    @Override
public void otorgarPuntos(int posicion, boolean vueltaRapida) {
    if (!posicionValida(posicion)) return;
    
    switch (posicion) {
        case 1 -> cantidadPuntos += 10;
        case 2 -> cantidadPuntos += 8;
        case 3 -> cantidadPuntos += 7;
        case 4 -> cantidadPuntos += 6;
        case 5 -> cantidadPuntos += 5;
        case 6 -> cantidadPuntos += 4;
        case 7 -> cantidadPuntos += 3;
        case 8 -> cantidadPuntos += 2;
        case 9 -> cantidadPuntos += 1;
        default -> cantidadPuntos += 0;
    }
}
public F2 subirCategoria(boolean tieneSuperlicencia) {
    return new F2(nombrePiloto, nacionalidad, numeroPlaza, escuderia, cantidadPuntos, tieneSuperlicencia);
    }

    @Override
        public String toString() {
        return super.toString() + " | Academia: " + nombreAcademia;
    }
}

File: Formula1/bin/Formula1/PHP/F3.php
<?php
namespace Formula1;

class F3 extends Monoplaza {
    private $nombreAcademia;

     ////////CONSTRUCTOR////////
    public function __construct(string $nombrePiloto, string $nacionalidad, int $numeroPlaza, string $escuderia, int $cantidadPuntos, string $nombreAcademia) {
        parent::__construct($nombrePiloto, $nacionalidad, $numeroPlaza, $escuderia, $cantidadPuntos);
        $this->nombreAcademia = $nombreAcademia;
    }

     // GETTERS Y SETTERS ////
    public function getNombreAcademia(){
    return $this->nombreAcademia;
    }
    public function setNombreAcademia($nombreAcademia){
    $this->nombreAcademia = $nombreAcademia;
}

 // METODOS ///
    public function posicionValida($posicion){
        if ($posicion < 1) {
        return false;
    }
    if ($posicion > 30) {
        return false;
    }
    return true;
    }

    public function otorgarPuntos($posicion,$vueltaRapida){
        if ($posicion < 1 || $posicion > 30) {
        return false;
        }
        switch ($posicion) {
        case 1:
            $this->cantidadPuntos += 10;
            break;
        case 2:
            $this->cantidadPuntos += 8;
            break;
        case 3:
            $this->cantidadPuntos += 7;
            break;
        case 4:
            $this->cantidadPuntos += 6;
            break;
        case 5:
            $this->cantidadPuntos += 5;
            break;
        case 6:
            $this->cantidadPuntos += 4;
            break;
        case 7:
            $this->cantidadPuntos += 3;
            break;
        case 8:
            $this->cantidadPuntos += 2;
            break;
        case 9:
            $this->cantidadPuntos += 1;
            break;
        default:
            break;
        }
    }

    public function subirCategoria($superLicencia){
        return new F2($this->nombrePiloto, $this->nacionalidad, $this->numeroPlaza, $this->escuderia, $this->cantidadPuntos, $superLicencia);
    }

    ///TOSTRING///
    public function __toString(){
    return parent::__toString() . "Academia:" . $this->nombreAcademia;
}
}